prime-router/src/test/kotlin/fhirengine/engine/FhirReceiverFilterTests.kt

```diff
@@ -0,0 +1,1054 @@
+                }
+        }
+
+        assertThat(actionLogger.logs).hasSize(2)
+        actionLogger.logs.forEach {
+            assertThat(it.detail)
+                .isInstanceOf<FHIRReceiverFilter.ReceiverItemFilteredActionLogDetail>()
+                .matchesPredicate { it.filterType == ReportStreamFilterType.PROCESSING_MODE_FILTER }
+        }
+    }
+
+    @Test
+    fun `fail - condition filter fails`() {
+        val fhirData = File(VALID_FHIR_URL).readText()
+
+        mockkObject(BlobAccess)
+
+        // set up
+        val actionLogger = ActionLogger()
+
+        // filters
+        val jurisFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val qualFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val routingFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val processingModeFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val conditionFilter = listOf(PROVENANCE_COUNT_EQUAL_TO_TEN)
+        val mappedConditionFilter = emptyList<ConditionFilter>()
+
+        val settings = FileSettings().loadOrganizations(
+            createOrganizationWithFilteredReceivers(
+                jurisFilter, qualFilter, routingFilter, processingModeFilter, conditionFilter, mappedConditionFilter
+            )
+        )
+
+        val engine = spyk(makeFhirEngine(metadata, settings) as FHIRReceiverFilter)
+        val messages = settings.receivers.map {
+            spyk(
+                FhirReceiverFilterQueueMessage(
+                    UUID.randomUUID(),
+                    BLOB_URL,
+                    "test",
+                    BLOB_SUB_FOLDER_NAME,
+                    topic = Topic.FULL_ELR,
+                    it.fullName
+                )
+            )
+        }
+
+        val bodyFormat = Report.Format.FHIR
+        val bodyUrl = BODY_URL
+
+        every { BlobAccess.uploadBlob(any(), any()) } returns "test"
+        every { accessSpy.insertTask(any(), bodyFormat.toString(), bodyUrl, any()) }.returns(Unit)
+
+        messages.forEach { message ->
+            every { message.downloadContent() }.returns(fhirData)
+            // act
+            accessSpy.transact { txn ->
+                val results = engine.run(message, actionLogger, actionHistory, txn)
+                assertThat(results).isEmpty()
+            }
+        }
+
+        azureEventService.getEvents().forEach { event ->
+            assertThat(event)
+                .isInstanceOf<ReportRouteEvent>()
+                .matchesPredicate {
+                    it.receiver == null
+                }
+        }
+        assertThat(actionLogger.logs).hasSize(2)
+        actionLogger.logs.forEach {
+            assertThat(it.detail)
+                .isInstanceOf<FHIRReceiverFilter.ReceiverItemFilteredActionLogDetail>()
+                .matchesPredicate { it.filterType == ReportStreamFilterType.CONDITION_FILTER }
+        }
+    }
+
+    @Test
+    fun `fail - mapped condition filter fails`() {
```

Comment:
Can we add a condition filter + mapped condition filter scenario?

Reply:
Great idea 💯 , added ✅ .

I also refactored / cleaned up these tests (and the ones for the prior step) to avoid carrying over a bunch of gunk.